### DISCUSSION TITLE
Change login module

### DISCUSSION THREAD

**Discussion by Gotiap** [oqtane/oqtane.framework#1186] [Q&A]
Hello everyone,
I need to create an external module for login. Let's say I have one application running where all users already there and need them to have an ability to login into oqtane application. So I am expecting that to be an external module, Can anyone help me with this?

Thank you for your time to read my question and helping me.

**Comment by hishamco** [comment]
This is doable, but I think it need some work to make this happen, more improtant is there should be a mapping between your users - which are in your storage - and Oqtane's users, because there might be other features that are rely on them

**Reply by hishamco** [reply]
Have a look to https://github.com/oqtane/oqtane.framework/blob/dev/Oqtane.Client/Modules/Admin/Login/Index.razor

**Reply by dchiragj** [reply]
Thank you for your time @hishamco ,

I tried to replicate that as a module. What I did is cloned that page and added some logic into the service that if that user does not exist in Oqtane database, It will add that user into Oqtane database, But I can not invoke authenticationstateprovider.

As you can see in the screenshot it shows null when I try to getservice of IdentityAuthenticationStateProvider. And also 
```
<AuthorizeView>
        <NotAuthorized>
        </NotAuthorized>
</AuthorizeView>

```
is not working as expected. So maybe something wrong with module-level authentication.

![image](https://user-images.githubusercontent.com/19463823/111459808-3236ac80-8741-11eb-923f-0824b495e0ac.png)

Can you please help me with this?

Thank you for your help.

**Reply by hishamco** [reply]
Please share a minimal repo so I can try what you did

**Reply by Gotiap** [reply]
Hello @hishamco 
[github repo](https://github.com/Gotiap/externalloginoctane)
Please find this repo.

Thank you.

**Reply by hishamco** [reply]
Seems you didn't register `ExternalLoginService`, I don't see any Startup class to register your customized services
